app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class ListsController < ApplicationController
+  before_action :authenticate_user!, :set_list, except: %i[index new create]
+
+  def index
+    @lists = List.page(params[:page])
+  end
+
+  def new
+    @list = List.new
+  end
+
+  def create
+    @list = List.new(permitted_attributes)
+
+    if @list.save
+      flash[:success] = "Your list \"#{@list.name}\" - created"
+      redirect_to @list
+    else
+      flash.now[:warning] = 'Invalid list parameters!'
+      render :new
+    end
+  end
+
+  def update
+    ActiveRecord::Base.transaction do
+      @list.update!(permitted_attributes)
+      BookManager.call(@list, params[:list][:book_ids])
```

Comment:
you had a specific requirement for a book-authors association: create an author in case of a new author name
there is no such request for a list-books creation. btw, event in that case, you didn't care about book authors

P.S. look at `accepts_nested_attributes`